client.go

```diff
@@ -85,13 +85,23 @@ func (c *Client) sendRequest(req *http.Request, v interface{}) error {
 		return fmt.Errorf("error, status code: %d, message: %w", res.StatusCode, errRes.Error)
 	}
 
-	if v != nil {
-		if err = json.NewDecoder(res.Body).Decode(v); err != nil {
+	return decodeResponse(v, *res)
+}
+
+func decodeResponse(v interface{}, res http.Response) error {
+	if v == nil {
+		return nil
+	}
+
+	if result, ok := v.(*string); ok {
```

Comment:
The `ok { ... }` block is rather large here, while the json decoding is a one-liner. Maybe invert the logic like:

```go
stringResponsePtr, isStringPtr := v.(*string)
if !isStringPtr {
	return json.NewDecoder(res.Body).Decode(v)
}

// ... rest of the logic 
```

Reply:
I agree, but I broke out the string decoding into its own function instead. The main reason being that the string handling is the exception, so should be handled immediately before we drop to the default handling.

This should mean if we have other values in the future which require special handling (e.g. an `os.File` or something similar), we can handle that before using our default `json` handling.